src/client/background.rs

```diff
@@ -0,0 +1,56 @@
+use futures::{Future, Poll};
+use hyper::body::Payload;
+use hyper::client::conn::Connection as HyperConnection;
+use std::fmt::{self, Debug};
+use tokio_io::{AsyncRead, AsyncWrite};
+
+use log::error;
+
+/// Background task for a client connection.
+///
+/// This type is not used directly by a user of this library,
+/// but it can show up in trait bounds of generic types.
+pub struct Background<T, B>
+where
+    T: AsyncRead + AsyncWrite + Send + 'static,
+    B: Payload,
+{
+    connection: HyperConnection<T, B>,
+}
+
+impl<T, B> Debug for Background<T, B>
```

Comment:
nit: can you move this to the bottom, i find the other impl blocks a lot more important and thus deserve to be higher up.

Reply:
OK. Personally, I prefer to keep manual `Debug` impls closer to the definition, to notice quicker if any newly added fields are missing from the output.